### DISCUSSION TITLE
Problem when use multiple direct on 1 template tag

### DISCUSSION THREAD

**Discussion by lyquocphong** [alpinejs/alpine#3509] [1. Help]
Hi !

I am new to alpinejs and I have face the weird problem when using multiple directive for template. Example:

```
<template x-if="canShow" x-for="source in sources">
  <div x-text="source.name"></div>
</template>
```

1. Is it possible to use multiple directive in the same template, in the example above x-for and x-if
2. I think the each directive will be handle differently, it mean alpinejs will handle the above example template 2 times, first time
it will handle for x-if and when canShow return true, it will try to render inside element and in that case the source is undefined 
and that is main issue I faced now. And it work just fine when handle x-for. Because I always get the first item as undefined
3. If we use the function for x-if or x-show, will it be reactive ?
4. If we set handler for event like onclick, the best practice is set it with or without "()". Example @click="sendEmail" or @click="sendEmail()", which one is the right one

**Comment by lyquocphong** [comment]
For some reason the example is missing, i will put here:
```
 <template x-if="canShow" x-for="source in showProductSources"
            >
<div
                                class="w-3 h-3 mr-1 mt-1 rounded-full"
                                :class="source.qty > 0 ? 'bg-green-500' : 'bg-yellow-400'"
                            >                            
 </template>
```

**Reply by ekwoka** [reply]
You cannot. Well, multiple directives does work, but these two would cause conflicting actions since they would each try to do different things.

what you can do is, in the x-for use a ternary

So in your example do

```
x-for="source in canShow ? showProductSources : []"
```

This would be the simplest way that doesn't use more elements just for this purpose.
